From: Leonid Bobrov <goleo@disroot.org>
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=disroot.org; s=mail;
	t=1592466144; bh=ta6+J4aYCs8Vhl2R/Us4erHImr9f617E2bNWoAWbI/I=;
	h=From:To:Cc:Subject:Date:In-Reply-To:References;
	b=koWmiyGumn3W2Hep8dVL3bG7SSQt5HUz8xsYmEoZCdnjJi3E096fWdpOTX2CQMTI/
	 qxg8M/HTmZqRk8sBtXBMQwtsNJYdU98v7bycR+A0aaujghAzwd7VbNfWyF5bOcdQ2S
	 sQcgZCF0gi4Pemxv80EhPcdqKtnrg+mksEiwGIzYYd/JpZ4RJzbvgmkxiBVqqCqhHV
	 iVtR5sMb/8zXppFbNUj6AACi3uJUtDRzi2I+pxJAvPoS/UUyiy7FP2OvvEQ5cfmqLe
	 Pa/1horcT2Iq51OBXTUOXQ82xiv13B3g7aS5w5Idx2kMD6KYk9nPkyuBcF6P8fst1s
	 mr6MBwuLKjR9Q==
To: ~sircmpwn/wio@lists.sr.ht,
	Leon Plickat <leonhenrik.plickat@stud.uni-goetttingen.de>
Cc: Leonid Bobrov <goleo@disroot.org>
Subject: [PATCH 09/11] Resize views by dragging their borders like in Rio
Date: Thu, 18 Jun 2020 10:39:52 +0300
Message-Id: <20200618073954.189606-10-goleo@disroot.org>
In-Reply-To: <20200618073954.189606-1-goleo@disroot.org>
References: <20200618073954.189606-1-goleo@disroot.org>
Mime-Version: 1.0
Content-Transfer-Encoding: quoted-printable

Like a glove!
---
 view.c | 36 ++++--------------------------------
 1 file changed, 4 insertions(+), 32 deletions(-)

diff --git a/view.c b/view.c
index fc7ccc7..c768cc4 100644
--- a/view.c
+++ b/view.c
@@ -216,39 +216,11 @@ struct wlr_box which_box(struct wio_server *server)=
 {
 }
=20
 struct wlr_box canon_box(struct wio_server *server, struct wlr_box box) =
{
-	if (box.width < MINWIDTH) {
-		switch (server->interactive.view->area) {
-		case VIEW_AREA_BORDER_TOP_LEFT:
-		case VIEW_AREA_BORDER_LEFT:
-		case VIEW_AREA_BORDER_BOTTOM_LEFT:
-			if ((box.x - server->interactive.sx) < server->interactive.view->xdg_=
surface->surface->current.width) {
-				box.x -=3D MINWIDTH - box.width;
-			}
-			break;
-		default:
-			if (box.x < server->interactive.sx) {
-				box.x -=3D MINWIDTH - box.width;
-			}
-		}
-		box.width =3D MINWIDTH;
-	}
-	if (box.height < MINHEIGHT) {
-		switch (server->interactive.view->area) {
-		case VIEW_AREA_BORDER_TOP_LEFT:
-		case VIEW_AREA_BORDER_TOP:
-		case VIEW_AREA_BORDER_TOP_RIGHT:
-			if ((box.y - server->interactive.sy) < server->interactive.view->xdg_=
surface->surface->current.height) {
-				box.y -=3D MINHEIGHT - box.height;
-			}
-			break;
-		default:
-			if (box.y < server->interactive.sy) {
-				box.y -=3D MINHEIGHT - box.height;
-			}
-		}
-		box.height =3D MINHEIGHT;
+	static struct wlr_box cache;
+	if (box.width < MINWIDTH || box.height < MINHEIGHT) {
+		return cache;
 	}
-	return box;
+	return cache =3D box;
 }
=20
 struct wio_view *wio_view_at(struct wio_server *server, double lx, doubl=
e ly,
--=20